### ISSUE TITLE
[Bug]: "Almost always use this as the left hand side of a WhenAny call" might be misleading

### ISSUE THREAD

**Issue by SuperJMN** [reactiveui/website#835]
### Describe the bug 🐞

I've been following the recommendations in the docs regarding the usage of `WhenAny`  (https://www.reactiveui.net/docs/guidelines/framework/use-this-on-left-of-whenany.html) but, recently, we faced a severe memory leak. I did some research and the offending line causing it was like this:

`this.WhenAnyValue(x => x.Service.SomeProperty).BindTo(this, x => x.MyProperty);`

...where Service was a reference to a instance that was injected via constructor. You can check the original file [here](https://github.com/zkSNACKs/WalletWasabi/blob/1ae3a7ae4e128b9f30d2d15c558c16acbf102b3d/WalletWasabi.Fluent/Models/Wallets/CoinModel.cs#L33).

The whole team was pretty confused, because the docs especifically mention says this is the good way to proceed when we want to monitor externally created dependencies, regardless of their lifetime:
> "Almost always use this as the left hand side of a WhenAny call"
> "Why? The lifetime of dependency is unknown - if it is a singleton it could introduce memory leaks into your application."

**I started to wonder if calling WhenAnyValue on `this` to watch *"foreign"* objects without explicitly disposing subscriptions was safe.**

**If it's not safe, then the docs are misleading**: It looks like watching properties of other dependencies (that can potentially have bigger lifetimes) doesn't need to manage subscriptions explicitly.

### Reproduction repository

https://github.com/reactiveui/ReactiveUI

### Expected behavior

The docs should be clear about disposing subscriptions to avoid memory leaks.

**Comment by anaisbetts** [comment]
It's still good advice, but it needs some additional qualifiers, if you alias a passed-in variable:

```csharp
public Service SomeOtherThing
public MyCoolViewModel(Service someOtherThing) {
  SomeOtherThing = someOtherThing;
  
  // ...
  
  this.WhenAny(x => x.SomeOtherThing.Foo).Subscribe();
}
```

You've reached through an object that was given to you, but made it look like you didn't! In that case, you need to treat it as if you just wrote:

```csharp
someOtherThing.WhenAny(....)
```

(i.e. you need to think about subscription lifetimes and Disposing)

**Comment by SuperJMN** [comment]
> It's still good advice, but it needs some additional qualifiers, if you alias a passed-in variable:
> 
> ```cs
> public Service SomeOtherThing
> public MyCoolViewModel(Service someOtherThing) {
>   SomeOtherThing = someOtherThing;
>   
>   // ...
>   
>   this.WhenAny(x => x.SomeOtherThing.Foo).Subscribe();
> }
> ```
> 
> You've reached through an object that was given to you, but made it look like you didn't! In that case, you need to treat it as if you just wrote:
> 
> ```cs
> someOtherThing.WhenAny(....)
> ```
> 
> (i.e. you need to think about subscription lifetimes and Disposing)

Yes, that makes sense, but then how is this working? https://github.com/SuperJMN-Tutorials/RxUI-MemoryLeak-Test?
Please, notice that this minimal repro doesn't have any memory leak, but it doesn't do any subscription management. 
- The `Item` class subscribes to a Service that is a Singleton. 
- A new list of Items are injected in the list when you click the Button and they seem to be freed up and GC's propertly! 🤔🤔🤔

**Comment by glennawatson** [comment]
Thanks for the careful write-up. The guidance itself is correct: using 'this' on the LHS avoids the singleton-holds-the-view-model leak. But you're right that the page didn't say loudly enough that you still need to dispose subscriptions when the right-hand side is rooted in a longer-lived dependency. The current branch adds a 'Caveat: still dispose your subscriptions' section to use-this-on-left-of-whenany.md covering WhenActivated/DisposeWith and the ToProperty exception.
